From 1bb21ea1797f3523fbda80be1156821b212231d7 Mon Sep 17 00:00:00 2001
From: Jerry Ray <jerry.ray@microchip.com>
Date: Tue, 10 Jun 2025 10:59:28 -0500
Subject: [linux-mchp][PATCH] ARM: configs: at91: sam9x75 Add switch support

The sam9x75 curiosity has a connector allowing it to support daughter cards
containing both single-port ethernet PHYs and ethernet switches. When an
ethernet switch is attached, the necessary DSA drivers and packages need to be
included in the kernel. This patch allows the SAM9x75 board to support the
KSZ8795, KSZ8895, KSZ9477, KSZ9563, and LAN9646 EDS2 boards.

Signed-off-by: Jerry Ray <jerry.ray@microchip.com>
---
 arch/arm/configs/at91_dt_defconfig | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/configs/at91_dt_defconfig b/arch/arm/configs/at91_dt_defconfig
index b4e16ca387f5..2fd2babffb36 100644
--- a/arch/arm/configs/at91_dt_defconfig
+++ b/arch/arm/configs/at91_dt_defconfig
@@ -44,6 +44,9 @@ CONFIG_IP_PNP=y
 CONFIG_IP_PNP_DHCP=y
 # CONFIG_INET_DIAG is not set
 CONFIG_IPV6_SIT_6RD=y
+CONFIG_BRIDGE=m
+CONFIG_NET_DSA=m
+CONFIG_VLAN_8021Q=m
 CONFIG_CAN=y
 CONFIG_BT=y
 CONFIG_BT_RFCOMM=y
@@ -90,6 +93,13 @@ CONFIG_BLK_DEV_SD=y
 # CONFIG_BLK_DEV_BSG is not set
 # CONFIG_SCSI_LOWLEVEL is not set
 CONFIG_NETDEVICES=y
+CONFIG_NET_DSA_MICROCHIP_KSZ_COMMON=m
+CONFIG_NET_DSA_MICROCHIP_KSZ9477_I2C=m
+CONFIG_NET_DSA_MICROCHIP_KSZ_SPI=m
+CONFIG_NET_DSA_MICROCHIP_KSZ_PTP=y
+CONFIG_NET_DSA_MICROCHIP_KSZ8863_SMI=m
+CONFIG_NET_DSA_SMSC_LAN9303_I2C=m
+CONFIG_NET_DSA_SMSC_LAN9303_MDIO=m
 # CONFIG_NET_VENDOR_BROADCOM is not set
 CONFIG_MACB=y
 CONFIG_DM9000=y
-- 
2.43.0

